pkg/services/libraryelements/database.go

```diff
@@ -251,13 +251,17 @@ func (l *LibraryElementService) deleteLibraryElement(c context.Context, signedIn
 		}
 
 		// then find the dashboards that were supposed to be connected to this element
-		_, requester := identity.WithServiceIdentity(c, signedInUser.GetOrgID())
-		dashs, err := l.dashboardsService.FindDashboards(c, &dashboards.FindPersistedDashboardsQuery{
-			Type:         searchstore.TypeDashboard,
-			OrgId:        signedInUser.GetOrgID(),
-			DashboardIds: dashboardIDs,
-			SignedInUser: requester, // a user may be able to delete a library element but not read all dashboards. We still need to run this check, so we don't allow deleting elements if dashboards are connected
+		// a identity may be able to delete a library element but not read all dashboards so we fetch then as the
+		// service user so we can prevent deletion of those connections
+		dashs, err := identity.WithServiceIdentityFn(c, signedInUser.GetOrgID(), func(ctx context.Context, ident identity.Requester) ([]dashboards.DashboardSearchProjection, error) {
+			return l.dashboardsService.FindDashboards(ctx, &dashboards.FindPersistedDashboardsQuery{
+				Type:         searchstore.TypeDashboard,
+				OrgId:        ident.GetOrgID(),
+				DashboardIds: dashboardIDs,
+				SignedInUser: ident,
+			})
```

Comment:
Why not just?
```suggestion
		newCtx, requester := identity.WithServiceIdentity(c, signedInUser.GetOrgID())
		dashs, err := l.dashboardsService.FindDashboards(newCtx, &dashboards.FindPersistedDashboardsQuery{
			Type:         searchstore.TypeDashboard,
			OrgId:        signedInUser.GetOrgID(),
			DashboardIds: dashboardIDs,
			SignedInUser: requester,
```

Reply:
Yeah I was thinking about that too. It is simpler so will change to that